Mathlib/AlgebraicTopology/SimplicialSet/KanComplex.lean

```diff
@@ -16,22 +16,20 @@ we show that every Kan complex is a quasicategory.
 ## TODO
 
 - Show that the singular simplicial set of a topological space is a Kan complex.
-- Generalize the definition to higher universes.
-  Since `Λ[n, i]` is an object of `SSet.{0}`,
-  the current definition of a Kan complex `S`
-  requires `S : SSet.{0}`.
 
 -/
 
+universe u
+
 namespace SSet
 
 open CategoryTheory Simplicial
 
 /-- A simplicial set `S` is a *Kan complex* if it satisfies the following horn-filling condition:
-for every `n : ℕ` and `0 ≤ i ≤ n`,
+for every nonzero `n : ℕ` and `0 ≤ i ≤ n`,
 every map of simplicial sets `σ₀ : Λ[n, i] → S` can be extended to a map `σ : Δ[n] → S`. -/
-class KanComplex (S : SSet) : Prop where
-  hornFilling : ∀ ⦃n : ℕ⦄ ⦃i : Fin (n+1)⦄ (σ₀ : Λ[n, i] ⟶ S),
-    ∃ σ : Δ[n] ⟶ S, σ₀ = hornInclusion n i ≫ σ
+class KanComplex (S : SSet.{u}) : Prop where
+  hornFilling : ∀ ⦃n : ℕ⦄ ⦃i : Fin (n + 2)⦄ (σ₀ : Λ[n + 1, i] ⟶ S),
+    ∃ σ : Δ[n + 1] ⟶ S, σ₀ = hornInclusion (n + 1) i ≫ σ
```

Comment:
Alternatively, we can add `n \ne 0` as assumption, instead of add `+1` everywhere.
Do you have a preference? Does it matter for certain defeqs?

Reply:
I prefer the `n + ...` solution, as this is something that can be inferred/pattern-matched automatically. This is consistent also with the `QuasiCategory.Basic` file (and my wip code on anodyne extensions).